### PR TITLE
Update GitHub templates and docs

### LINKED ISSUE #5436: .github directory cleanup issue templates

### LINKED ISSUE #5438: Update the ./.github md files to have correct monorepo guidance

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Cleans up the `.github` directory with the following updates:

- Update issue templates
- Update `CONTRIBUTING.md` to fit new monorepo structure

Closes #5438
Closes #5436

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -80,34 +80,24 @@ Each contributor is required to [sign a CLA](https://cla.shopify.com/). This pro
...
 
 Making sure the voice and tone is consistent across Polaris is important to us, for more information visit the [voice and tone guidelines](https://polaris.shopify.com/content/voice-and-tone).
 
 ### Development workflow
 
 After cloning Polaris React, run `yarn` to fetch its dependencies. Then you can run several commands:
 
-- `yarn dev` runs a Storybook server which includes a playground editable at `playground/Playground.tsx`
+- `yarn dev:react` runs a Storybook server which includes a playground editable at `playground/Playground.tsx`
 - `yarn test` runs the complete test suite
```
Comment (sam-b-rose) [inline .github/CONTRIBUTING.md]: We still need to add a root-level test command. But, I decided to keep this as-is for now since we plan to add it very soon (this week) in #5433

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -80,34 +80,24 @@ Each contributor is required to [sign a CLA](https://cla.shopify.com/). This pro
...
 - `yarn test <pattern>` runs tests with matching filenames
-- `yarn tophat` runs a local development server and mounts all of the component examples
 
 We recommend running `yarn test` (or its variations above), as well as trying your build of Polaris React in a real project, to make sure you don’t introduce any regressions as you work on your change.
 
-Run `yarn run build-consumer <project>` where `<project>` is the directory the build will be copied. **The `<project>` directory must be a sibling of the `polaris-react` directory**.
```
Comment (sam-b-rose) [inline .github/CONTRIBUTING.md:102]: Will create an issue to remove this `build-consumer` command as I don't know how much value it adds to our project.

Should we just remove it or would it be helpful to provide a `yalc`-based alternative for testing changes in a separate project? My thoughts are that it is out of scope for our documentation and recommendations.

@kyledurand @aveline

Comment (sam-b-rose) [reply .github/CONTRIBUTING.md:102]: I did just [stumble across this](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md#how-to--documentation) where the `build-consumer` command is used to test Component README changes in the `polaris-styleguide`.

Seems a bit unclear based on the steps how this all works and will likely change in the near future when we migrate documentation to the monorepo.

Are there alternate ways to test documentation updates you have tried (like `yalc`) or is this still the best solution?

Comment (kyledurand) [reply .github/CONTRIBUTING.md:102]: `build-consumer` is super useful for testing unreleased changes in web or also on the styleguide. I think it was added before yalc was a thing so updating it to use yalc would probably be the right move here.

Comment (aaronccasanova) [reply .github/CONTRIBUTING.md:102]: Agreed, +1 for updating the docs to demonstrate a `yalc` based workflow

Comment (sam-b-rose) [reply .github/CONTRIBUTING.md:102]: I've added some basic guidance on using `yalc` to test package changes, but would like to [do some more clean up](https://github.com/Shopify/polaris/issues/5550) and detailed tophatting instructions for `spin` as well.

I've broken that out into it's own task since it involves removing scripts and creating new doc files 📜➕➕➕

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -80,33 +80,76 @@ Each contributor is required to [sign a CLA](https://cla.shopify.com/). This pro
...
+Use the [documentation directory](https://github.com/Shopify/polaris/tree/main/documentation) for guidance on: [accessibility testing](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md) and [adding to the changelog](https://github.com/Shopify/polaris/blob/main/documentation/Versioning%20and%20changelog.md). For code style best practices, take a look at existing components.
 
 Making sure the voice and tone is consistent across Polaris is important to us, for more information visit the [voice and tone guidelines](https://polaris.shopify.com/content/voice-and-tone).
 
 ### Development workflow
 
-After cloning Polaris React, run `yarn` to fetch its dependencies. Then you can run several commands:
+After cloning Polaris, run `yarn` to fetch its dependencies. Then you can run several commands:
```
Comment (sam-b-rose) [inline .github/CONTRIBUTING.md:93]: @aaronccasanova, @kyledurand  I plan to continue to iterate on these docs after addressing #5433 ✍️ 

Anything blocking this merge?

Comment (aaronccasanova) [reply .github/CONTRIBUTING.md:93]: Yalc docs looks great 🙌

**Review by aaronccasanova — APPROVED** [review]
